lib/graphlient/adapters/faraday_adapter.rb

```diff
@@ -0,0 +1,37 @@
+require 'faraday'
+
+module Graphlient
+  module Adapters
+    class FaradayAdapter
+      attr_accessor :url, :headers
+
+      def initialize(url, headers: {})
+        @url = url
+        @headers = headers
+      end
+
+      def execute(document:, operation_name:, variables:, context:)
+        response = conn.post do |req|
+          req.headers['Content-Type'] = 'application/json'
+          req.body = {
+            query: document.to_query_string,
+            operationName: operation_name,
+            variables: variables.to_json
+          }.to_json
+        end
+        JSON.parse(response.body)
```

Comment:
For Faraday you should be using Faraday middleware to parse JSON and set content type, like [here](https://github.com/codegram/hyperclient/blob/49bff2d6c6fa8bb502cc3b3e3f5b8417a735c855/lib/hyperclient/entry_point.rb#L141).